.github/workflows/rust.yml

```
@@ -39,13 +40,23 @@ jobs:
           available_crates="$(find . -name Cargo.toml -printf '%h\n' | sed 's:./::' | sort | tac)"
           changed_files="$(git diff --name-only HEAD^ HEAD)"
 
-          changed_crates=""
+          # All jobs for all crates will run if any of those paths changes
```

```suggestion
          # All jobs for all crates will run if any of these paths changes
```

if you're talking about `(".github" "rust-toolchain.toml")` below

---

Do we want to also do it if `cargo.toml` changes or no need?

---

How come `Cargo.toml` causes the jobs to run for all creates but the `rust-toolchain.toml` doesn't, I'm a little confused. 

Is this logic we've put elsewhere? And if so can we combine it?

---

The last time I changed `rust-toolchain.toml`, it had an issue with `rustfmt`. changing `Cargo.toml` ensures dependent crates are compiled, but by changing the file could not introduce other errors I think

---

I think changing `Cargo.toml` could introduce errors, no? What if we change the default features argument for example?